packages/optimizer/lib/transformers/ReorderHeadTransformer.js

```diff
@@ -50,24 +50,24 @@ class HeadNodes {
   }
 
   _removeDuplicateCustomExtensions(extensions) {
-    const nodesByName = new Map();
+    const nodesBySrc = new Map();
     for (const node of extensions) {
-      const name = this._getName(node);
-      nodesByName.set(name, node);
+      const src = node.attribs['src'];
+      nodesBySrc.set(src, node);
     }
-    return Array.from(nodesByName.values());
+    return Array.from(nodesBySrc.values());
   }
```

Comment:
I took an alternative approach to this in PHP, to do the deduplication in the `registerScript` function itself: https://github.com/ampproject/amp-toolbox-php/pull/130/files#diff-b0da2cc892f74bf1ebc7a84e63d7065894b24541a48f3a26089f23860ee1209fR156-R190

This also accounts for `module` and `nomodule` variations.